watertap/tools/oli_api/tests/test_credentials.py

```diff
@@ -8,59 +8,95 @@
+# You are under no obligation whatsoever to provide any bug fixes, patches, or upgrades to the
+# features, functionality or performance of the source code ("Enhancements") to anyone; however,
+# if you choose to make your Enhancements available either publicly, or directly to OLI Systems, Inc.,
+# without imposing a separate written license agreement for such Enhancements, then you hereby grant
+# the following license: a non-exclusive, royalty-free perpetual license to install, use, modify, prepare
+# derivative works, incorporate into other computer software, distribute, and sublicense such enhancements
+# or derivative works thereof, in binary and source code form.
+###############################################################################
 
 import pytest
-from os import remove
 
-from watertap.tools.oli_api.credentials import CredentialManager, cryptography_available
+from os.path import join
+from pathlib import Path
+from os import listdir, remove
 
-# from watertap.tools.oli_api.client import OLIApi
+from watertap.tools.oli_api.client import OLIApi
+from watertap.tools.oli_api.credentials import (
+    CredentialManager,
+    cryptography_available,
+)
 
 
 @pytest.fixture
 def credential_manager():
     if not cryptography_available:
-        pytest.skip(reason="cryptography module not available")
+        pytest.skip(reason="cryptography module not available.")
     credentials = {
-        "username": "dummy_username@email.com",
-        "password": "dummy_password",
-        "root_url": "https://dummyrooturl.com",
-        "auth_url": "https://dummyauthurl.com",
+        "access_keys": [""],
+        "root_url": "",
+        "auth_url": "",
     }
-    return CredentialManager(**credentials, test=True)
+    try:
+        credential_manager = CredentialManager(**credentials, test=True)
+        yield credential_manager
+    except:
+        pytest.xfail("Unable to test OLI logins.")
 
 
-@pytest.mark.unit
-def test_encryption(credential_manager):
-    key = credential_manager.encryption_key
-    assert (
-        CredentialManager(encryption_key=key, test=True).credentials
-        == credential_manager.credentials
-    )
-    remove(credential_manager.config_file)
+@pytest.fixture
+def oliapi_instance(credential_manager):
+    def _cleanup(file_list, file_path):
+        new_files = [
+            f"{file_path}/{f}" for f in listdir(file_path) if f not in file_list
+        ]
+        for f in new_files:
+            remove(f)
+
+    root_dir = Path(__file__).parents[1]
+    test_dir = Path(__file__).parents[0]
+    root_contents = listdir(root_dir)
+    test_contents = listdir(test_dir)
+
+    credential_manager.login()
+    with OLIApi(credential_manager, test=True) as oliapi:
+        local_dbs_file = join(test_dir, "test.dbs")
+        oliapi.get_dbs_file_id(local_dbs_file)
+        yield oliapi
+
+    _cleanup(root_contents, root_dir)
+    _cleanup(test_contents, test_dir)
```

Comment:
Since it's not being used in this test file and we have the fixture defined in `conftest.py`, can this be deleted?

Reply:
Yes, will push my changes to the test.credentials file shortly